src/BuiltInTools/dotnet-watch/DotNetWatcher.cs

```diff
@@ -54,6 +39,21 @@ public override async Task WatchAsync(CancellationToken cancellationToken)
                     return;
                 }
 
+                StaticFileHandler? staticFileHandler;
+                ProjectGraphNode? projectRootNode;
+                if (evaluationResult.ProjectGraph != null)
+                {
+                    projectRootNode = evaluationResult.ProjectGraph.GraphRoots.Single();
+                    var projectMap = new ProjectNodeMap(evaluationResult.ProjectGraph, Context.Reporter);
+                    staticFileHandler = new StaticFileHandler(Context.Reporter, projectMap, browserConnector);
+                }
+                else
+                {
+                    Context.Reporter.Verbose("Unable to determine if this project is a webapp.");
```

Comment:
What's this condition? Why is it tied to web apps?

Reply:
This code path is legacy mode (not Hot Reload). If we can't load the project graph we can't detect project capabilities and can't do browser refresh.